### PR TITLE
Feature/GCP CIS 1.1 Full Support

### PR DESCRIPTION
**Pull request by x4v13r64**
# Description

This PR implements all the checks from GCP CIS 1.1.

## Type of change

Select the relevant option(s):

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works (optional)
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: ScoutSuite/providers/gcp/rules/findings/cloudsql-instances-public-ips.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        "https://cloud.google.com/resource-manager/docs/organization-policy/org-policy-constraints",
+        "https://console.cloud.google.com/iam-admin/orgpolicies/sql-restrictPublicIp"
+    ],
+    "dashboard_name": "Instances",
+    "path": "cloudsql.projects.id.instances.id",
+    "conditions": [
+        "and",
+        [
+            "cloudsql.projects.id.instances.id.public_ip",
+            "notEmpty",
```
Comment (j4v) [inline ScoutSuite/providers/gcp/rules/findings/cloudsql-instances-public-ips.json:24]: @SophieDorval this is being triggered when the value is `None`:

sc_2021-04-12_12h35m21s

File: ScoutSuite/providers/gcp/resources/iam/bindings_separation_duties.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        super().__init__(facade)
+        self.project_id = project_id
+
+    async def fetch_all(self):
+        raw_bindings = await self.facade.cloudresourcemanager.get_member_bindings(self.project_id)
+        binding_id, binding = await self._parse_binding_separation(raw_bindings)
+        self[binding_id] = binding
+        x=1
+
+    async def _parse_binding_separation(self, raw_bindings):
```
Comment (j4v) [inline ScoutSuite/providers/gcp/resources/iam/bindings_separation_duties.py:16]: @SophieDorval this is missing the `name` field, so the partial shows up with a blank name.

**Comment by j4v** [comment]
@SophieDorval & @xnkevinnguyen can you resolve the merge conflicts please?

**Comment by SophieDorval** [comment]
@j4v Fixed the conflict.

**Comment by alessandrogonzalez** [comment]
Hi,

I was talking to someone that was testing this branch, and the following error was found during execution:

`
2021-04-28 10:38:09 NCC-laptop asyncio[2115765] ERROR Task exception was never retrieved
future: <Task finished name='Task-40' coro=<MonitoringAlertPolicies.fetch_all() done, defined at <REDACTED>/ScoutSuite/ScoutSuite/providers/gcp/resources/stackdrivermonitoring/monitoring_alert_policies.py:10> exception=AttributeError("'google.protobuf.pyext._message.RepeatedCompositeCo' object has no attribute '_values'")>
Traceback (most recent call last):
  File "<REDACTED>/ScoutSuite/ScoutSuite/providers/gcp/resources/stackdrivermonitoring/monitoring_alert_policies.py", line 12, in fetch_all
    alert_policy = self._parse_alert_policy(raw_alert_policies)
  File "<REDACTED>/ScoutSuite/ScoutSuite/providers/gcp/resources/stackdrivermonitoring/monitoring_alert_policies.py", line 19, in _parse_alert_policy
    self._specific_alert_policy_present(raw_alert_policies)
  File "<REDACTED>/ScoutSuite/ScoutSuite/providers/gcp/resources/stackdrivermonitoring/monitoring_alert_policies.py", line 32, in _specific_alert_policy_present
    for condition in alert_policy.conditions._values:
AttributeError: 'google.protobuf.pyext._message.RepeatedCompositeCo' object has no attribute '_values'
`

Is it possible that the `_values` property is not always present and we are assuming it is? or maybe we don't need the `._values` at all?

Note that this error is also present in the v6 PR.

**Comment by x4v13r64** [comment]
> Note that this error is also present in the v6 PR.

@alessandrogonzalez I see this is now fixed, correct?

**Comment by alessandrogonzalez** [comment]
> > Note that this error is also present in the v6 PR.
> 
> @alessandrogonzalez I see this is now fixed, correct?

It is fixed in this PR, so it should get fixed when this is merged and develop branch is merged into v6 branch.

**Comment by x4v13r64** [comment]
Weird, looks like there are commits to the https://github.com/nccgroup/ScoutSuite/tree/feature/974-gcp-cis branch that aren't showing up in this PR.

**Comment by x4v13r64** [comment]
@alessandrogonzalez & @lowSoA is there a timeline or blocker to merge this PR into `develop`? It adds a bunch of functionality and we've been using it for some time now.

**Review by lowSoA — APPROVED** [review]

**Review by alessandrogonzalez — APPROVED** [review]
LGTM

**Pull request merged**